### PR TITLE
General Map and Traversal functions over context and Sexpressions

### PR DESCRIPTION
**Pull request by mariari**

### CODE REVIEW

File: library/StandardLibrary/src/Juvix/Library/Sexp.hs [hunk 1]
```diff
@@ -111,9 +153,9 @@ nameFromT (Atom (A name _)) = Just name
 nameFromT _ = Nothing
 
 assoc :: T -> T -> Maybe T
-assoc t (car :> cdr)
-  | t == car = Just (cadr car)
-  | otherwise = assoc t cdr
+assoc t (car' :> cdr')
+  | t == car car' = Just (cadr car')
```
Comment (cyberglot) [inline library/StandardLibrary/src/Juvix/Library/Sexp.hs:157]: wait, `cadr`? is that correct? 🤔  am I missing something?

**Comment by cyberglot** [comment]
I have just one comment, otherwise LGTM.

**Review by cyberglot — COMMENTED** [review]
Just have one question, otherwise LGTM.

File: library/StandardLibrary/src/Juvix/Library/Sexp.hs (hunk 1 above)
Comment (mariari) [reply library/StandardLibrary/src/Juvix/Library/Sexp.hs:157]: yeah, as we are searching a list like

((:a 3) (:b 4)) 

so if we do

assoc :a then we should get 3 back so

(cadar *) ===>  3.

we have an extra a as we are doing this cadr on the car thus forming cadar

**Review by cyberglot — APPROVED** [review]

**Review by cwgoes — COMMENTED** [review]
ACK

**Pull request merged**